### ISSUE TITLE
Unable to reproduce Camelyon16 results

### ISSUE THREAD

**Issue by Yew-yy** [szc19990412/TransMIL#25]
@szc19990412 
Thank you very much for your excellent work, it has given me a lot of inspiration. However I had some problems reproducing the results mentioned in your paper. I browsed through the previous issues and I trained the camelyon16 data exactly the way you mentioned, including using a 20x patch, setting patience=10 or 30, and dividing the training and validation sets in a 10:1 ratio.
Nevertheless, on the test set I obtained similar results to those mentioned in issue 4. The results are far from those mentioned in the paper. Is there any training detail that I missed, if so, please let me know, thanks a lot.
test_Accuracy=0.7109375
test_CohenKappa=0.37618548,
test_Recall=0.6844485
test_Precision=0.6931818
auc=0.6600362
![1677579858756](https://user-images.githubusercontent.com/81496304/221826908-38aac4cb-ac33-4605-a639-b23569bcdaa5.png)

**Comment by yyyyxxxyyyy** [comment]
Dear ：

I hope this message finds you well. I am writing to you because I am a beginner in MIL and I am having some difficulty understanding it. I came across your review and I was wondering if you could provide me with some guidance on how to reproduce this code.

Would you mind sharing with me the detailed steps that you took to reproduce the code? I am particularly interested in understanding the process from start to finish, including any necessary pre-processing steps, data selection, and any other important details that may be relevant.

I appreciate any assistance you can provide and I look forward to hearing back from you soon.

Best regards

**Comment by yyyyxxxyyyy** [comment]
> @szc19990412 Thank you very much for your excellent work, it has given me a lot of inspiration. However I had some problems reproducing the results mentioned in your paper. I browsed through the previous issues and I trained the camelyon16 data exactly the way you mentioned, including using a 20x patch, setting patience=10 or 30, and dividing the training and validation sets in a 10:1 ratio. Nevertheless, on the test set I obtained similar results to those mentioned in issue 4. The results are far from those mentioned in the paper. Is there any training detail that I missed, if so, please let me know, thanks a lot. test_Accuracy=0.7109375 test_CohenKappa=0.37618548, test_Recall=0.6844485 test_Precision=0.6931818 auc=0.6600362 ![1677579858756](https://user-images.githubusercontent.com/81496304/221826908-38aac4cb-ac33-4605-a639-b23569bcdaa5.png)
Dear ：

I hope this message finds you well. I am writing to you because I am a beginner in MIL and I am having some difficulty understanding it. I came across your review and I was wondering if you could provide me with some guidance on how to reproduce this code.

Would you mind sharing with me the detailed steps that you took to reproduce the code? I am particularly interested in understanding the process from start to finish, including any necessary pre-processing steps, data selection, and any other important details that may be relevant.

I appreciate any assistance you can provide and I look forward to hearing back from you soon.

Best regards

**Comment by lingxitong** [comment]
> @szc19990412 Thank you very much for your excellent work, it has given me a lot of inspiration. However I had some problems reproducing the results mentioned in your paper. I browsed through the previous issues and I trained the camelyon16 data exactly the way you mentioned, including using a 20x patch, setting patience=10 or 30, and dividing the training and validation sets in a 10:1 ratio. Nevertheless, on the test set I obtained similar results to those mentioned in issue 4. The results are far from those mentioned in the paper. Is there any training detail that I missed, if so, please let me know, thanks a lot. test_Accuracy=0.7109375 test_CohenKappa=0.37618548, test_Recall=0.6844485 test_Precision=0.6931818 auc=0.6600362 ![1677579858756](https://user-images.githubusercontent.com/81496304/221826908-38aac4cb-ac33-4605-a639-b23569bcdaa5.png)

maybe some tricks had not shown in the paper

**Comment by weiaicunzai** [comment]
I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.

**Comment by lingxitong** [comment]
> I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.

I can get 0.87,should use early stop

**Comment by weiaicunzai** [comment]
> > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> 
> I can get 0.87,should use early stop

Thanks, glad to hear. Did you set `shuffle`  in the class `CamelData`  to True during training? I didn't noticed that until now, the default shuffle argument is False  ` data_shuffle: False`  in the file `TransMIL.yaml`.

**Comment by lingxitong** [comment]
i use the origin setting&nbsp;



---Original---
From: ***@***.***&gt;
Date: Sat, Jan 6, 2024 16:59 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25)




   
I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
  
I can get 0.87,should use early stop
  
Thanks, glad to hear. Did you set shuffle in the class CamelData  to True during training? I didn't noticed that until now, the default shuffle argument is False   data_shuffle: False  in the file TransMIL.yaml.

**Comment by weiaicunzai** [comment]
> i use the origin setting&nbsp;
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Sat, Jan 6, 2024 16:59 PM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25) I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper. I can get 0.87,should use early stop Thanks, glad to hear. Did you set shuffle in the class CamelData to True during training? I didn't noticed that until now, the default shuffle argument is False data_shuffle: False in the file TransMIL.yaml. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

Thanks,  could you please tell me how do you process the `Camelyon16 ` dataset? I think the performance gap between our results might be caused by the different way of processing the training data.  The default  settings of CLAM api doesn't perform well in the Camelyon16 dataset . some slides can not even generate patches, So I use the training data released by [DTFD-MIL](https://github.com/hrzhang1123/DTFD-MIL) to train trans-mil.

**Comment by lingxitong** [comment]
lingxitong vchat



---Original---
From: ***@***.***&gt;
Date: Sat, Jan 6, 2024 17:19 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25)




  
i use the origin setting&nbsp;
 …
 ---Original--- From: @.&gt; Date: Sat, Jan 6, 2024 16:59 PM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25) I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper. I can get 0.87,should use early stop Thanks, glad to hear. Did you set shuffle in the class CamelData to True during training? I didn't noticed that until now, the default shuffle argument is False data_shuffle: False in the file TransMIL.yaml. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;
  
Thanks,  could you please tell me how do you process the Camelyon16  dataset? I think the performance gap between our results might be caused by the different way of processing the training data.  The default  settings of CLAM api doesn't perform well in the Camelyon16 dataset . some slides can not even generate patches, So I use the training data released by DTFD-MIL to train trans-mil.

**Comment by weiaicunzai** [comment]
> I can get 0.87,should use early stop



> lingxitong vchat
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Sat, Jan 6, 2024 17:19 PM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25) i use the origin setting&nbsp; … ---Original--- From: @.&gt; Date: Sat, Jan 6, 2024 16:59 PM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [szc19990412/TransMIL] Unable to reproduce Camelyon16 results(Issue #25) I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper. I can get 0.87,should use early stop Thanks, glad to hear. Did you set shuffle in the class CamelData to True during training? I didn't noticed that until now, the default shuffle argument is False data_shuffle: False in the file TransMIL.yaml. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt; Thanks, could you please tell me how do you process the Camelyon16 dataset? I think the performance gap between our results might be caused by the different way of processing the training data. The default settings of CLAM api doesn't perform well in the Camelyon16 dataset . some slides can not even generate patches, So I use the training data released by DTFD-MIL to train trans-mil. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

I'm sorry, but what do you mean by 'vchat' ?  You manually labeled the tissue areas yourself?

**Comment by weiaicunzai** [comment]
> > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> 
> I can get 0.87,should use early stop

After setting shuffle to True, I'm getting acc around 0.84. I believe 0.87 is very feasible.

**Comment by lingxitong** [comment]
> > > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> > 
> > 
> > I can get 0.87,should use early stop
> 
> After setting shuffle to True, I'm getting acc around 0.84. I believe 0.87 is very feasible.

wechat:lingxitong

**Comment by weiaicunzai** [comment]
> vchat



> i use the origin setting&nbsp;



> > > > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> > > 
> > > 
> > > I can get 0.87,should use early stop
> > 
> > 
> > After setting shuffle to True, I'm getting acc around 0.84. I believe 0.87 is very feasible.
> 
> wechat:lingxitong

Thanks, I have sent the friend request.

**Comment by homadnmr** [comment]
> @szc19990412 Thank you very much for your excellent work, it has given me a lot of inspiration. However I had some problems reproducing the results mentioned in your paper. I browsed through the previous issues and I trained the camelyon16 data exactly the way you mentioned, including using a 20x patch, setting patience=10 or 30, and dividing the training and validation sets in a 10:1 ratio. Nevertheless, on the test set I obtained similar results to those mentioned in issue 4. The results are far from those mentioned in the paper. Is there any training detail that I missed, if so, please let me know, thanks a lot. test_Accuracy=0.7109375 test_CohenKappa=0.37618548, test_Recall=0.6844485 test_Precision=0.6931818 auc=0.6600362 ![1677579858756](https://user-images.githubusercontent.com/81496304/221826908-38aac4cb-ac33-4605-a639-b23569bcdaa5.png)

HI as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by homadnmr** [comment]
> > vchat
> 
> > i use the origin setting
> 
> > > > > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> > > > 
> > > > 
> > > > I can get 0.87,should use early stop
> > > 
> > > 
> > > After setting shuffle to True, I'm getting acc around 0.84. I believe 0.87 is very feasible.
> > 
> > 
> > wechat:lingxitong
> 
> Thanks, I have sent the friend request.

HI as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by homadnmr** [comment]
> > > > I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.
> > > 
> > > 
> > > I can get 0.87,should use early stop
> > 
> > 
> > After setting shuffle to True, I'm getting acc around 0.84. I believe 0.87 is very feasible.
> 
> wechat:lingxitong

Hi :)  as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by homadnmr** [comment]
HI as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by homadnmr** [comment]
> I can get around 0.78 - 0.79 acc, however, still 10 percent less than reported in the paper.

Hi as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by homadnmr** [comment]
> @szc19990412 Thank you very much for your excellent work, it has given me a lot of inspiration. However I had some problems reproducing the results mentioned in your paper. I browsed through the previous issues and I trained the camelyon16 data exactly the way you mentioned, including using a 20x patch, setting patience=10 or 30, and dividing the training and validation sets in a 10:1 ratio. Nevertheless, on the test set I obtained similar results to those mentioned in issue 4. The results are far from those mentioned in the paper. Is there any training detail that I missed, if so, please let me know, thanks a lot. test_Accuracy=0.7109375 test_CohenKappa=0.37618548, test_Recall=0.6844485 test_Precision=0.6931818 auc=0.6600362 ![1677579858756](https://user-images.githubusercontent.com/81496304/221826908-38aac4cb-ac33-4605-a639-b23569bcdaa5.png)

Hi :)  as I can see you could run the code. May I ask you to share the exact environment package versions?

**Comment by undercutspiky** [comment]
Hi, so were you able to reproduce the results? I've been unable to reproduce the results for most of these MIL methods...they all seem to have around 80% accuracy on the test set except TransMIL which is at best around 70 like your results (and issue 4). Have you found any tricks to make the accuracy go up? I've tried different hyperparameters and also gradient accumulation.
